ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -141,8 +175,19 @@ class HttpRequestData internal constructor(
     val headers: Headers,
     val body: OutgoingContent,
     val executionContext: Job,
-    val attributes: Attributes
+    val attributes: Attributes,
+    @KtorExperimentalAPI
+    val extensions: Map<KType, HttpRequestExtension>
```

Comment:
Could we use `AttributeKey` as a key instead of `KType`?

Reply:
Good idea, I replaced `KType` on `AttributeKey.`